doc/read-the-docs-site/adr/0006-indexer-resuming-strategy.rst

```diff
@@ -45,16 +45,88 @@ Decision
-  However, the ``Coordinator`` will make sure that all indexers are synced until a common point, and advance at the same speed (i.e. they can only request the next block once all the indexers have finished processing the current block).
+* We will start a single node chain-sync client for all indexers instead of one node chain sync client per indexer.
 
-* We will use the full set of points provided by ``resumableFromStorage`` of an indexer as resuming points for the node-to-client chain-sync protocol.
-  These points need to ordered in descending order so that protocol can priotise the selection of the latest ones.
+* We will implement the following resuming logic.
+
+  In order to start the chain-sync client, we need to provide to provide intersection points (or resuming points).
+  Intersection points represent the points from which we want to start syncing from the local node.
+  We will provide as intersection points the points returned from the following pseudo-code:
+
+  .. code-block:: haskell
+
+    -- | Query the resumable points of each indexer then select the resumable
+    -- points of the indexer that has the lowest immutable point. If two indexer
+    -- have the same lowest immutable point, select the one which has the lowest
+    -- overall point (immutable or not).
+    selectIntersectionPoints indexers = do
+      resumingPoints <- mapM queryResumingPoints indexers
+      head $ sortOn (\rps -> (maximum $ filter isImmutablePoint rps, maximum rps)) resumingPoints
+
+  Next, we need a function which processes each ``ChainSyncEvent`` returned by the chain-sync client for each indexer.
+  The pseudo-code is as follows:
+
+  .. code-block:: haskell
+
+    applyChainSyncEvent event indexer = do
+      case event of
+        lastSyncPoints <- getLastSyncPoints indexer
+        Resume point -> do
+          maybePoint <- getPointFromSlotNo (slotNo point) indexer
+          case maybePoint of
+            Nothing ->
+              pure () -- Do nothing
+            Just _ ->
+              rewind point indexer
+        RollBackward point -> do
+          rewind point indexer
+        RollForward block point -> do
+          lastSyncPoints <- getLastSyncPoints indexer
+          let lastSyncSlotNos = fmap getSlotNo lastSyncPoints
+          let lowestLastSyncSlotNo = minimum lastSyncSlotNos
+          let highestLastSyncSlotNo = maximum lastSyncSlotNos
+
+          -- Block already processed. Do nothing
+          if point `elem` lastSyncPoints
+          then do
+            pure ()
+
+          -- SlotNo of new block point already processed, but with different block header hash
+          else if getSlotNo point `elem` lastSyncSlotNos
+          then do
+            -- TODO Not exactly correct, we need to rewind to `point - 1`, but we don't necessarly have
+            -- the block header hash of that point
+            rewind point indexer
```

Comment:
it doesn't include the idea of tracking the point of the last traced event we discussed, isn't it?

Reply:
I forgot to commit :)

Reply:
Can you revalidate now?

Reply:
Yes, all good.